internal/lookoutv2/pruner/pruner_test.go

```diff
@@ -0,0 +1,168 @@
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/armadaproject/armada/internal/common/compress"
+	"github.com/armadaproject/armada/internal/common/database/lookout"
+	"github.com/armadaproject/armada/internal/common/util"
+	"github.com/armadaproject/armada/internal/lookoutingesterv2/instructions"
+	"github.com/armadaproject/armada/internal/lookoutingesterv2/lookoutdb"
+	"github.com/armadaproject/armada/internal/lookoutingesterv2/metrics"
+	"github.com/armadaproject/armada/internal/lookoutv2/repository"
+)
+
+var baseTime, _ = time.Parse("2006-01-02T15:04:05.000Z", "2022-03-01T15:04:05.000Z")
+
+func TestPruneDb(t *testing.T) {
+	type testJob struct {
+		jobId string
+		ts    time.Time
+	}
+
+	type testCase struct {
+		testName    string
+		expireAfter time.Duration
+		jobs        []testJob
+		jobIdsLeft  []string
+	}
+
+	nIds := 100
+	sampleJobIds := make([]string, nIds)
+	for i := 0; i < nIds; i++ {
+		sampleJobIds[i] = util.NewULID()
+	}
+
+	manyJobs := func(startIdx, endIdx int, ts time.Time) []testJob {
+		var testJobs []testJob
+		for i := startIdx; i < endIdx; i++ {
+			testJobs = append(testJobs, testJob{
+				jobId: sampleJobIds[i],
+				ts:    ts,
+			})
+		}
+		return testJobs
+	}
+
+	testCases := []testCase{
+		{
+			testName:    "no jobs",
+			expireAfter: 10 * time.Hour,
+			jobs:        []testJob{},
+			jobIdsLeft:  []string{},
+		},
+		{
+			testName:    "no expired jobs",
+			expireAfter: 10 * time.Hour,
+			jobs: []testJob{
+				{
+					jobId: sampleJobIds[0],
+					ts:    baseTime,
+				},
+				{
+					jobId: sampleJobIds[1],
+					ts:    baseTime.Add(-9 * time.Hour),
+				},
+			},
+			jobIdsLeft: []string{sampleJobIds[0], sampleJobIds[1]},
+		},
+		{
+			testName:    "expire a job",
+			expireAfter: 10 * time.Hour,
+			jobs: []testJob{
+				{
+					jobId: sampleJobIds[0],
+					ts:    baseTime,
+				},
+				{
+					jobId: sampleJobIds[1],
+					ts:    baseTime.Add(-9 * time.Hour),
+				},
+				{
+					jobId: sampleJobIds[2],
+					ts:    baseTime.Add(-(10*time.Hour + 1*time.Minute)),
+				},
+			},
+			jobIdsLeft: []string{sampleJobIds[0], sampleJobIds[1]},
+		},
+		{
+			testName:    "expire many jobs",
+			expireAfter: 100 * time.Hour,
+			jobs: util.Concat([][]testJob{
+				manyJobs(0, 10, baseTime.Add(-300*time.Hour)),
+				manyJobs(10, 20, baseTime.Add(-200*time.Hour)),
+				manyJobs(20, 50, baseTime.Add(-(100*time.Hour + 5*time.Minute))),
+				manyJobs(50, 100, baseTime.Add(-99*time.Hour)),
+			}),
+			jobIdsLeft: sampleJobIds[50:],
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.testName, func(t *testing.T) {
+			err := lookout.WithLookoutDb(func(db *pgxpool.Pool) error {
+				converter := instructions.NewInstructionConverter(metrics.Get(), "armadaproject.io/", &compress.NoOpCompressor{})
+				store := lookoutdb.NewLookoutDb(db, metrics.Get(), 3, 10)
+
+				ctx := context.TODO()
```

Comment:
I reckon put a timeout here- that way if  the unit test ends up triggering some crazy database loop won't hang.